core/src/test/java/io/github/tcdl/msb/impl/RequesterImplTest.java

```diff
@@ -185,6 +186,31 @@ public void testRequestMessage() throws Exception {
         assertNotNull(requestMessage.getRawPayload());
     }
 
+    @Test
+    public void testRequestMessageWithTags() throws Exception {
```

Comment:
add more tests please:
where the original message provides tags, and new one is added
where the original message provides tag A, and we add tags X,A. We get X,A or A, in outgoing requests